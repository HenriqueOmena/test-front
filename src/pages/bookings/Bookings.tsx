import { format } from "date-fns";
import React from "react";
import { useLocation } from "react-router";
import { Link } from "react-router-dom";
import BookingsStyle from "./Bookings.style";
import CheckCircleIcon from "@material-ui/icons/CheckCircle";
import { Button } from "@material-ui/core";

const Bookings = () => {
  const { state } = useLocation<{ name: string; start: string }>();
  const classes = BookingsStyle();
  return (
    <div className={classes.successCard}>
      <CheckCircleIcon fontSize="large" className={classes.successIcon} />
      <h3>Appointment booked Successfully!</h3>
      <p>
        Appointment booked with <strong>{state?.name}</strong>
        <br /> on <strong>{state && format(new Date(state?.start), "d LLL yyyy hh:mma")}</strong>
      </p>
      <Link to="/" className={classes.link}>
        <Button variant="contained" color="primary">
          Back To Home
        </Button>
      </Link>
    </div>
  );
};

export default Bookings;
