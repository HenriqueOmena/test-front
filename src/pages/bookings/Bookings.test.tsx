import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import routeData from "react-router";
import Bookings from "./Bookings";

const mockLocation = {
  pathname: "/welcome",
  hash: "",
  search: "",
  state: { name: "some name", start: "2021-08-26T05:49:16.334Z" }
};

describe("Component - Bookings", () => {
  let component = (
    <MemoryRouter>
      <Bookings />
    </MemoryRouter>
  );

  beforeEach(() => {
    jest.spyOn(routeData, "useLocation").mockReturnValue(mockLocation);
  });

  it("Should Match SnapShot", () => {
    expect(component).toMatchSnapshot();
  });

  it("Should render with prop name", async () => {
    const { getByText } = render(component);
    const nameText = getByText("Appointment booked Successfully!");

    expect(nameText).toBeVisible();
  });
});
