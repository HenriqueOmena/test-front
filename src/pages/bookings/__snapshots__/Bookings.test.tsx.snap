// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Bookings Should Match SnapShot 1`] = `
<MemoryRouter>
  <Bookings />
</MemoryRouter>
`;
